### PR TITLE
Sub generator upgrade: personnal code is deleted

### PR DESCRIPTION
**Pull request by lordlothar99**
Documentation for #3696

### CODE REVIEW

**Comment by lordlothar99** [comment]
I faced problems with ruby and jekyll on this computer... I was unable to test properly; I have to reinstall a bunch of stuff, but anyway I posted the PR to avoid delays...
I included great graph made by @PierreBesson , with some updates of my own, matching new upgrade process ; unfortunately, GitGraph is unable to display orphan branches, but it doesn't matter that much. Let me know what you think about the whole

**Comment by PierreBesson** [comment]
I think somereview of everything should be done. Also Gitgraph should be added as a js lib included in the website not using the jsfiddle.

I know testing Jekyll can be a pain, that's why I always test it using  docker as explained in the Repo's readme.

**Comment by pascalgrimaud** [comment]
@PierreBesson : not sure the Docker image of Jekyll is still working... The last time (~2 weeks ago) I tried, it failed ! Can you try it ?

**Comment by PierreBesson** [comment]
@pascalgrimaud, I just tested it and it's not working. I got it to work using another image:
`docker run --rm --label=jekyll --volume=$(pwd):/site -p 4000:4000 andredumas/github-pages serve --watch`

**Comment by jdubois** [comment]
We are hosted by GitHub pages, so I'm not sure if we can have GitGraph working anyway. I'd much rather just have the images, we are sure it will work without any issues.

File: pages/upgrading_an_app.md [hunk 1]
```diff
@@ -25,23 +26,41 @@ To upgrade your application, type:
...
+8. Re-generate the application using the `yo jhipster --force --with-entities` command.
+9. Commit the generated code to the `jhipster_upgrade` branch.
+10. Merge the `jhipster_upgrade` branch back to the original branch from where the `yo jhipster:upgrade` command was launched.
+11. Now you just need to proceed with resolving merge conflicts if there are any.
 
 Congratulations, your application is now upgraded with the latest version of JHipster!
+
+![GitGraph]{{ site.url }}/images/upgrade_gitgraph.png)
+
+## Advice
```
Comment (jdubois) [inline pages/upgrading_an_app.md]: this doesn't render on my machine, I'm having a look at it

Comment (lordlothar99) [inline pages/upgrading_an_app.md]: I finally have jekyll working (thx to Docker), and I have the same problem. I'm having a look

Comment (jdubois) [inline pages/upgrading_an_app.md]: I'm correcting it all, no worries

Comment (jdubois) [inline pages/upgrading_an_app.md]: I need to do the merge myself, as I'll release the doc just after the generator

Comment (lordlothar99) [inline pages/upgrading_an_app.md]: oops, I just pushed the fix :smile:

**Comment by PierreBesson** [comment]
@jdubois, there is no problem to have gitgraph working in Github Pages. It's available as a 20K js frontend lib. https://github.com/nicoespeon/gitgraph.js/tree/develop/build
We just need to add something like this:
```
<canvas id="gitGraph"></canvas>
<script>
var master = gitGraph.branch( "master" );
gitGraph.commit( "Initial commit" );
gitGraph.commit( "My second commit" ).commit( "Add awesome feature" );
...
</script> 
```

Also @lordlothar99, when I look at your jsfiddle (http://jsfiddle.net/lordlothar99/tqp9gyu3), it doesn't match the image you pushed:

**Comment by lordlothar99** [comment]
thx @PierreBesson, I just forgot to publish latest version ; fixed now

**Comment by PierreBesson** [comment]
OK, let's first merge this with the image for today's release. Then when I have time I will convert it to use gitgraph.

**Comment by jdubois** [comment]
I've already merged this in my branch -> it should be OK, and I will push everything at the same time (I'm currently writing the release notes and doing final tests)

**Comment by jdubois** [comment]
I'm closing this as I merged it already on the website - I hope I didn't forgot anything!!

**Pull request closed without merging**